src/main.ts

```diff
@@ -0,0 +1,528 @@
+
+    if (this.backtrace !== null) {
+      execution_env.RUST_BACKTRACE = this.backtrace;
+    }
+    if (this.modify_profile !== null) {
+      if (this.modify_profile) {
+        execution_env.NIX_INSTALLER_MODIFY_PROFILE = "true";
+      } else {
+        execution_env.NIX_INSTALLER_MODIFY_PROFILE = "false";
+      }
+    }
+
+    if (this.nix_build_group_id !== null) {
+      execution_env.NIX_INSTALLER_NIX_BUILD_GROUP_ID = `${this.nix_build_group_id}`;
+    }
+
+    if (this.nix_build_group_name !== null) {
+      execution_env.NIX_INSTALLER_NIX_BUILD_GROUP_NAME =
+        this.nix_build_group_name;
+    }
+
+    if (this.nix_build_user_prefix !== null) {
+      execution_env.NIX_INSTALLER_NIX_BUILD_USER_PREFIX =
+        this.nix_build_user_prefix;
+    }
+
+    if (this.nix_build_user_count !== null) {
+      execution_env.NIX_INSTALLER_NIX_BUILD_USER_COUNT = `${this.nix_build_user_count}`;
+    }
+
+    if (this.nix_build_user_base !== null) {
+      execution_env.NIX_INSTALLER_NIX_BUILD_USER_ID_BASE = `${this.nix_build_user_count}`;
+    }
+
+    if (this.nix_package_url !== null) {
+      execution_env.NIX_INSTALLER_NIX_PACKAGE_URL = `${this.nix_package_url}`;
+    }
+
+    if (this.proxy !== null) {
+      execution_env.NIX_INSTALLER_PROXY = this.proxy;
+    }
+
+    if (this.ssl_cert_file !== null) {
+      execution_env.NIX_INSTALLER_SSL_CERT_FILE = this.ssl_cert_file;
+    }
+
+    if (this.diagnostic_endpoint !== null) {
+      execution_env.NIX_INSTALLER_DIAGNOSTIC_ENDPOINT =
+        this.diagnostic_endpoint;
+    }
+
+    // TODO: Error if the user uses these on not-MacOS
+    if (this.mac_encrypt !== null) {
+      execution_env.NIX_INSTALLER_ENCRYPT = this.mac_encrypt;
+    }
+
+    if (this.mac_case_sensitive !== null) {
+      execution_env.NIX_INSTALLER_CASE_SENSITIVE = this.mac_case_sensitive;
+    }
+
+    if (this.mac_volume_label !== null) {
+      execution_env.NIX_INSTALLER_VOLUME_LABEL = this.mac_volume_label;
+    }
+
+    if (this.mac_root_disk !== null) {
+      execution_env.NIX_INSTALLER_ROOT_DISK = this.mac_root_disk;
+    }
+
+    if (this.logger !== null) {
+      execution_env.NIX_INSTALLER_LOGGER = this.logger;
+    }
+
+    if (this.log_directives !== null) {
+      execution_env.NIX_INSTALLER_LOG_DIRECTIVES = this.log_directives;
+    }
+
+    // TODO: Error if the user uses these on MacOS
+    if (this.init !== null) {
+      execution_env.NIX_INSTALLER_INIT = this.init;
+    }
+
+    if (this.start_daemon !== null) {
+      if (this.start_daemon) {
+        execution_env.NIX_INSTALLER_START_DAEMON = "true";
+      } else {
+        execution_env.NIX_INSTALLER_START_DAEMON = "false";
+      }
+    }
+
+    let extra_conf = "";
+    if (this.github_token !== null) {
+      extra_conf += `access-tokens = github.com=${this.github_token}`;
+      extra_conf += "\n";
+    }
+    if (this.trust_runner_user !== null) {
+      // TODO: Consider how to improve this
```

Comment:
Improve what part? The necessity of trusted-users, or the `$USER` detection?

Reply:
I think this is fine, actually!